### ISSUE TITLE
Lint for mixing cascades and conditional operator

### ISSUE THREAD

**Issue by cbracken** [dart-lang/linter#86]
I suspect that the behaviour of the following is unexpected to many people:
```dart
var foo1 = new Foo()..bar = 1;
...
var foo2 = (foo1 != null)
    ? foo1
    : new Foo2()..bar = 2;
// foo1.bar == 2 here
```
Since the conditional operator has higher precendence than the cascade, the above is equivalent to `var foo2 = (foo1)..bar = 2;` which is not likely what the author intended, but which is free from errors/warnings.

**Comment by sgonyea** [comment]
I'd add `??` and `??=` to the list as well.

**Comment by MichaelRFairhurst** [comment]
I think the correct lint is: cascade on _anything_ but `=`, `.`, `?.`, `[]`, `new`, or another cascade.

Reason is simple: the cascade operator has a really low precedence, only higher than '='. (it might actually be even lower than = and rely on the fact that `a = b` evaluates to `b` to work....but in either case, = works as expected). It _looks_ like it has the same precedence as `.` (and `?.` and `[]` and `new`).

When `..` applies to `=`, the precedence is intuitive.
When `..` applies to `.` (and the like) it behaves the same as if they had the same precedence. (`a.b.c` and `a.b..c` are the same, precedence-wise).
When `..` applies to `..`, precedence doesn't matter because intuitively all operators have their own precedence.

But when `..` is applied to any other type of expression (+, -, &&, !, *, etc etc), the actual precedence of rock-bottom is likely not the expected precedence, that of `.`, and that difference in perspective _will_ have an impact.
